src/app/shared/user/user.service.ts

```diff
@@ -50,12 +56,16 @@ export class UserService {
    *
    * @param userId
    */
-  addUserToWorkflows(userId: number): void {
+  addUserToWorkflows(userId: number, entryType: EntryType): void {
     this.alertService.start('Adding user to existing workflows and tools on Dockstore');
     this.usersService.addUserToDockstoreWorkflows(userId).subscribe(
       (workflows: Array<Workflow>) => {
         this.alertService.detailedSuccess();
-        this.workflowService.setWorkflows(workflows);
+        if (entryType === EntryType.Tool) {
+          this.mytoolsService.getMyEntries(userId, entryType);
+        } else {
+          this.myWorkflowsService.getMyEntries(userId, entryType);
```

Comment:
> For the `else` condition, you already have the `workflows` from line 56/62, and you could just do a filter in Typescript instead of making another (relatively) expensive network call.

Not currently possible, because the `addUserToDockstoreWorkflows` endpoint only returns the user's `BioWorkflows`, so there's no `Notebooks` or other stuff to filter for.

I like the two-step "'add' and then 'get'" mechanism.  Sure, there's a little extra latency, but it seems to separate the concerns better and only retrieves entries of the necessary type.  If we stuck with it, I'd propose that we [eventually?] change the `addUserToDockstoreWorkflows` endpoint to not return anything.

Reply:
Ah, you're right, I saw the method returns `Workflow` and assumed it would return all subclasses of `Workflow`, but it does only return `BioWorkflow`.

Reply:
Followup to my followup, you could just use `workflows` if the entry type is BioWorkflow and not make the extra call. But for symmetry, maybe it's best to keep it as is.

> I'd propose that we [eventually?] change the addUserToDockstoreWorkflows endpoint to not return anything.

Not sure I agree with that one, but we can cross that bridge if we get there.

Reply:
> Followup to my followup, you could just use `workflows` if the entry type is BioWorkflow and not make the extra call.

I don't know if this happens for others, but on my end (locally), when the UI uses the `workflows` response instead of `getMyEntries`, clicking "Discover Existing Dockstore Workflows" doesn't display the newly synced workflows right away:

![Screenshot from 2024-04-03 16-05-37](https://github.com/dockstore/dockstore-ui2/assets/122565245/32053e3f-ca0b-4e1f-a4b4-86168e9e0b74)

If I refresh the page though, all the expected workflows repopulate the page/sidebar. Could be a personal issue... does anyone else encounter something like this? (noting that I had zero private workflows prior to calling the endpoint)

Reply:
Do you mean this happens with your fix or without your fix? Or both?

Also, you should be able to see the network response in the browser dev tools; does that look correct?

Reply:
For me it happens without the fix, with the original `this.workflowService.setWorkflows(workflows);` line. Network looks okay too (request to /api/users/{id}/workflows, 200 status code, response is the array of `BioWorkflow` entries).

Reply:
I'll give it a shot locally when I get a chance, but I may be delayed.

Reply:
> on my end (locally), when the UI uses the `workflows` response instead of `getMyEntries`, clicking "Discover Existing Dockstore Workflows" doesn't display the newly synced workflows right away:

(referring to the original code without the fix) Hmm, I think this might be an edge case for users that don't have any private workflows. I tried registering a test workflow before clicking "Discover Existing Dockstore Workflows", and everything appeared to display correctly.I guess it's because, when a user goes from zero to 1+ workflows via syncing, the UI needs to reprocess the workflow to be selected/displayed?)

If this is the case, I think it might be ideal to just always make the second request. With the `getMyEntries` fix, I didn't notice these issues in the UI, even though I had zero private workflows prior to syncing.

Reply:
> I guess it's because, when a user goes from zero to 1+ workflows via syncing, the UI needs to reprocess the workflow to be selected/displayed?

Hmm, not sure, but that's a good theory. But yeah, it's probably best to keep the the code the same for all entry types.